### PR TITLE
Formatting Ranges: Range "m" specifier do not pass through to underlying element

### PR DESCRIPTION
**Pull request by Mick235711**
In libc++ implementation of C++23 ranges formatting facility, the following code (MVP):
```cpp
#include <format>
#include <vector>
#include <iostream>

int main()
{
    std::vector<std::pair<int, int>> vs{{1, 1}, {2, 2}};
    std::cout << std::format("vs = {0}\nmap = {0:m}\nmapmap = {0:m:m}\n", vs);
    return 0;
}
```
outputs ([Compiler Explorer @ x86-64 clang trunk with libc++](https://godbolt.org/z/GPboxKKoY))
```
vs = [(1, 1), (2, 2)]
map = {(1, 1), (2, 2)}
mapmap = {1: 1, 2: 2}
```

However, the m specifier for ranges should also imply m formatting (i.e. "key: value" for pair and 2-tuple) on the elements of the ranges, as per [\[tab:formatter.range.type\]/1](https://eel.is/c++draft/tab:formatter.range.type):
```
Indicates that the opening bracket should be "{", the closing bracket should be "}", the separator should be ", ",
and each range element should be formatted as if m were specified for its tuple-type.
```
In other words, `{:m}` should be equivalent to `{:m:m}`, yet in the above output it clearly isn't.

### CODE REVIEW

**Comment by Mick235711** [comment]
Hmm, not really sure why the test fails here but pass in c++23. Is there anything changed on range formatting in c++2c?

**Comment by mordante** [comment]
Thanks for the patch!
As mentioned here https://github.com/llvm/llvm-project/issues/60995#issuecomment-1458656667 I wonder whether it is intended that `:m` overrides `n` of the underlying _tuple-type_ formatter. I want to discuss this with the Committee before looking at the details of this patch.

**Comment by Mick235711** [comment]
Well, that's an unlikely title for the same issue...😄
Sure, I guess this should be an LWG issue then.
